src/compiler/checker.ts

```diff
@@ -2517,7 +2522,7 @@ namespace ts {
                     return typeReferenceToTypeNode(<TypeReference>type);
                 }
                 if (type.flags & TypeFlags.TypeParameter || objectFlags & ObjectFlags.ClassOrInterface) {
-                    const name = symbolToName(type.symbol, context, SymbolFlags.Type, /*expectsIdentifier*/ false);
+                    const name = type.symbol ? symbolToName(type.symbol, context, SymbolFlags.Type, /*expectsIdentifier*/ false) : createIdentifier("?");

```

Comment:
When does this case occur?

Reply:
This happens when we have a synthetic type parameter, such as the type parameters we create for tuple types and the `markerXXXType` type parameters we use for variance determination.